compiler/rustc_resolve/src/late.rs

```diff
@@ -1559,12 +1559,15 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
     }
 
     fn with_mod_rib<T>(&mut self, id: NodeId, f: impl FnOnce(&mut Self) -> T) -> T {
-        let module = self.r.expect_module(self.r.local_def_id(id).to_def_id());
+        let def_id = self.r.local_def_id(id);
+        let module = self.r.expect_module(def_id.to_def_id());
         // Move down in the graph.
         let orig_module = replace(&mut self.parent_scope.module, module);
         self.with_rib(ValueNS, RibKind::Module(module), |this| {
             this.with_rib(TypeNS, RibKind::Module(module), |this| {
                 let ret = f(this);
+                this.r.doc_link_resolutions.entry(def_id).or_default();
+                this.r.doc_link_traits_in_scope.entry(def_id).or_default();
```

Comment:
Could you add a comment like "make sure every module has an entry in these tables, required by metadata encoding" to this logic? It looks strange otherwise.